### PR TITLE
Fix typos and bugs with UG, update UG

### LINKED ISSUE #104: [PE-D] inconsistency in command format
As shown in the document, there should not be a blank space between '-@' and '<date>'. However, in example there is a blank space. Thought it would not affect any uses. 

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: Raymond0212/ped#4

### LINKED ISSUE #106: [PE-D] Wrong example of edit command in UG
As attached, the example given in UG cannot be run well.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: XIA-LIYI/ped#2

### LINKED ISSUE #112: [PE-D] UG Bug: the term "API" might not be appropriate
UG Bug (Explanations, correctness): In the table of contents (Features), the usage of the word API may not be the appropriate term for the User Guide. An application programming interface is a computing interface which defines interactions between multiple software *intermediaries*, not to the end user. Additionally, the word might be confusing to the end user, which may or may not know what an API is. The term usage would be more approriate.

Steps to reproduce: Accessible from [UG](https://ay2021s1-cs2103-w14-3.github.io/tp/UserGuide.html)

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: theodoreleebrant/ped#1

### LINKED ISSUE #113: [PE-D] UG Bug (Explanations): Quick start may not be descriptive enough.
Kindly provide an instruction on how to start the application, e.g. by the command `java -jar PACKAGE_NAME` 

Steps to reproduce: Accessible from [UG](https://ay2021s1-cs2103-w14-3.github.io/tp/UserGuide.html#QuickStart)

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: theodoreleebrant/ped#2

### LINKED ISSUE #114: [PE-D] UG Bug (Neatness): Lack of spacing in between commands
No details provided.Consider adding more spacing in between each commands, as this would reduce cluttering.

Steps to reproduce: Accessible from [UG](https://ay2021s1-cs2103-w14-3.github.io/tp/UserGuide.html)

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: theodoreleebrant/ped#3

### LINKED ISSUE #115: [PE-D] UG Bug: Typo in Features > List
Currently on the UG: "Displays a list of **the all the** user’s expenses."

Steps to reproduce: Accessible from [UG](https://ay2021s1-cs2103-w14-3.github.io/tp/UserGuide.html#Features)

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: theodoreleebrant/ped#4

### LINKED ISSUE #116: [PE-D] UG Bug (Neatness): Typo in Fields > Note
Currently in the UG: Prefixes function with or without a whitespace after. (e.g. -d dinner and -ddinner will yield the same results) 

* There should be no "a" preceeding whitespace.
* There should be no full stop after the word "after"
* Bracketing error

Steps to reproduce: Accessible from [UG](https://ay2021s1-cs2103-w14-3.github.io/tp/UserGuide.html#Fields)

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: theodoreleebrant/ped#5

### LINKED ISSUE #126: [PE-D] Invalid command error is wrong
Error Message:

Invalid command format! 
add: Adds an expense to the expense book. Parameters: -d DESCRIPTION -$AMOUNT [-@DATE] [t/TAG]...
Example: add -dUniqlo Jacket -$59.90 -@04-10-2020 t/friends 

But in the UG, the t/ is for CATEGORY.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: theodoreleebrant/ped#15

### LINKED ISSUE #130: [PE-D] Small typo in user guide
Perhaps dd-mm-yyyy.

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: WangZijun97/ped#3

### LINKED ISSUE #135: [PE-D] UG edit example bug
I think there should be a spacing between '-d' and 'lunch'.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: timjkong/ped#2

### PR DESCRIPTION
**Pull request by maxxyh**
Fixes #135, fixes #130, fixes #126, fixes #116, fixes #115, fixes #114, fixes #113, fixes #112, fixes #106, fixes #104.

Added a command format guide
Add documentation for `clear` command
Changed all references of Tag to Category in user-displayed messages

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -2,117 +2,213 @@
...
 3. [Usage](#Usage)
 4. [Command Summary](#CommandSummary)
 
 ## Quick Start <a name="QuickStart"></a>
+> _Bamboo runs on Windows, OS-X, and Unix systems._
+1. Ensure that Java Version 11 or above is installed in your computer
+2. Download the latest Bamboo Release `ExpenseBook.jar` [here](https://github.com/AY2021S1-CS2103-W14-3/tp/releases/tag/v1.3).
+3. Copy the file to the folder you want to use as the home directory for Bamboo.
+4. Double-click the file to start the app. It should open in a new window.  
+    OR Navigate to the directory on terminal and run `java -jar <FILENAME>.jar`.
```
Comment (lebencwb) [inline docs/UserGuide.md]: I think we can specify this as `ExpenseBook.jar` instead of the placeholder `<FILENAME>.jar`

File: docs/UserGuide.md [hunk 2]
```diff
@@ -2,117 +2,213 @@
...
+5. Type in commands with arguements in the command input box above, and press `Enter` to execute them.
```
Comment (lebencwb) [inline docs/UserGuide.md]: There's a small typo here: 'arguments'

File: docs/UserGuide.md [hunk 3]
```diff
@@ -2,117 +2,213 @@
...
+    - Works only in complement with [add](#add), [edit](#edit), [find](#find), [switch](#switch), [addCat](#addCat), [deleteCat](#deleteCat), [topup](#topup)
+    - Format: `t/<category>`
+    - Example: `edit t/Food`, `find -d Coffee t/Food`
+
+1. **remark** <a name="remark"></a>
+    - Additional information about an expense.
+    - Works only in complement with [remark](#remark)
+    - Format: `-r <remark>`
+    - Restrictions: Remarks can be up to 200 characters long.
+    - Example: `edit t/Food`, `find -d Coffee t/Food`
```
Comment (lebencwb) [inline docs/UserGuide.md]: I think this example is inaccurate. Maybe `remark 1 -r impulse spending` or `remark 2 -r emergency` might be better examples?

**Review by lebencwb — CHANGES_REQUESTED** [review]
Just a few minor issues from me, other than that it's looking good!

File: docs/UserGuide.md (hunk 1 above)
Comment (maxxyh) [reply docs/UserGuide.md]: Fixed!

File: docs/UserGuide.md (hunk 2 above)
Comment (maxxyh) [reply docs/UserGuide.md]: Fixed!

File: docs/UserGuide.md (hunk 3 above)
Comment (maxxyh) [reply docs/UserGuide.md]: Fixed as well!

**Review by lebencwb — APPROVED** [review]
LGTM!

**Pull request merged**